internal/sessiontoken/sessiontoken.go

```diff
@@ -525,8 +525,21 @@ func (s *SessionToken) fetchSSOWebToken(clientID, awsFedAppID string, at *access
 	if err != nil {
 		return nil, err
 	}
+
 	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("fetching SSO web token received API response %q", resp.Status)
+		bodyBytes, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return nil, fmt.Errorf("fetching SSO web token received API response %q", resp.Status)
+		}
+
+		var apiErr apiError
+		err = json.NewDecoder(bytes.NewReader(bodyBytes)).Decode(&apiErr)
+		if err != nil {
+			return nil, fmt.Errorf("fetching SSO web token received API response %q", resp.Status)
+		}
+
+		return nil, fmt.Errorf("fetching SSO web token received API response %q\nerror: %q, description: %q\n",
```

Comment:
@emanor-okta the golang linter doesn't like newlines in error messages so perhaps listing least noisy information first would be the way to go for human readability

```go
return nil, fmt.Errorf("fetching SSO web token received API response %q, description: %q, error: %q", resp.Status, apiErr.ErrorDescription, apiErr.Error)
```